r-pkg/R/elasticsearch_eda_funs.R

```diff
@@ -114,3 +114,145 @@ get_counts <- function(field
+    
+    ######################### parse the result ###############################
+    # flatten the list object that is returned from the query
+    flattened <- unlist(mapping)
+    
+    # the names of the flattened object has the index, type, and field name
+    # however, it also has extra terms that we can use to split the name
+    # into three distinct parts
+    mappingCols <- stringr::str_split_fixed(names(flattened), '\\.(mappings|properties)\\.', n = 3)
+    
+    # convert to data table and add the data type column
+    mappingDT <- data.table::data.table(meta = mappingCols, data_type = as.character(flattened))
+    newColNames <- c('index', 'type', 'field', 'data_type')
+    data.table::setnames(mappingDT, old = names(mappingDT), new = newColNames)
+    
+    # remove any rows, where the field does not end in ".type" to remove meta info
+    mappingDT <- mappingDT[stringr::str_detect(field, '\\.type$')]
+    
+    # mappings in nested objects have sub-fields called properties
+    # mappings of fields that are indexed in different ways have multiple fields
+    # we want to remove these terms from the field name
+    metaRegEx <- '\\.(properties|fields|type)'
+    mappingDT[, field := stringr::str_replace_all(field, metaRegEx, '')]
+    
+    return(mappingDT)
+}
+
+# [title] Get a data.table containing names of indices and aliases
+# [es_host] A string identifying an Elasticsearch host.
+#' @importFrom httr content GET stop_for_status
+.get_aliases <- function(es_host) {
+    
+    # construct the url to the alias endpoint
+    url <- paste0(es_host, '/_cat/aliases')
+    
+    # make the request
+    result <- httr::GET(url = url)
+    httr::stop_for_status(result)
+    resultContent <- httr::content(result, as = 'text')
+    
+    if (is.null(resultContent)) {
+        # there are no aliases in this Elasticsearch cluster
+        return(NULL)
+    } else {
+        return(.process_alias(alias_string = resultContent))
+    }
+}
+
+# [title] Process the string returned by the GET alias API into a data table
+# [alias_string] A string returned by the alias API with index and alias name
+#' @importFrom stringr str_replace str_split
+#' @importFrom data.table as.data.table setnames
+.process_alias <- function(alias_string) {
+    # remove the new line at the end of the string, if it exists
+    aliasString <- stringr::str_replace(alias_string, '\n$', '')
+    
+    # split each entry, separated by a new line, into a vector in a list
+    aliases <- stringr::str_split(aliasString, '\n')[[1]]
+    
+    # remove white space and only take the first two entries
+    aliases <- stringr::str_split(aliases, '\\s+')
+    aliases <- lapply(aliases, function(pair) pair[1:2])
+    
+    # create a data table from the resulting list
+    aliasDT <- data.table::as.data.table(matrix(unlist(aliases), byrow = TRUE, ncol = 2))
+    data.table::setnames(aliasDT, old = colnames(aliasDT), new = c('alias', 'index'))
+    return(aliasDT)
```

Comment:
I think this whole function can be replaced by
```
aliasDT <- data.table::data.table(read.table(text = alias_string, stringsAsFactors = FALSE))
return(aliasDT[, .(alias = V1, index = V2)])
```
It's twice as fast on my machine!